include/event_manager.h

```diff
@@ -147,52 +144,142 @@ extern "C" {
+ *
+ * The event hook called when the event is being processed.
+ * The hook function should have a form `void hook(const struct event_header *eh)`.
+ * The macro makes sure that the hook provided here is called last.
+ * Only one hook can be registered with this macro.
+ *
+ * @param hook_fn Hook function
+ */
+#define EVENT_HOOK_PREPROCESS_REGISTER_LAST(hook_fn)                                        \
+	const struct {} __event_hook_preprocess_last_sub_redefined = {};                    \
+	_EVENT_HOOK_PREPROCESS_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_MARKER_LAST_ELEMENT))
 
-/** @brief Initialize tracing in the Event Manager.
- * The behavior of this function depends on the actual implementation.
- * The default implementation of this function is no-operation.
- * It is annotated as weak and is meant to be overridden by layer
- * adding support for profiling mechanism.
+/**
+ * @brief Register event hook on the end of event processing. that would be called first
  *
- * @retval 0 If the operation was successful. Error values can be added by
- *         overwriting this function.
- **/
-int event_manager_trace_event_init(void);
+ * The event hook called after he event is processed.
+ * The hook function should have a form `void hook(const struct event_header *eh)`.
+ * The macro makes sure that the hook provided here is called first.
+ * Only one hook can be registered with this macro.
+ *
+ * @param hook_fn Hook function
+ */
+#define EVENT_HOOK_POSTPROCESS_REGISTER_FIRST(hook_fn)                                        \
+	const struct {} __event_hook_postprocess_first_sub_redefined = {};                    \
+	_EVENT_HOOK_POSTPROCESS_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_MARKER_FIRST_ELEMENT))
+
+/**
+ * @brief Register event hook on the end of event processing.
+ *
+ * The event hook called after he event is processed.
+ * The hook function should have a form `void hook(const struct event_header *eh)`.
+ *
+ * @param hook_fn Hook function
+ */
+#define EVENT_HOOK_POSTPROCESS_REGISTER(hook_fn)                                          \
+	_EVENT_HOOK_POSTPROCESS_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_SUBS_PRIO_NORMAL))
+
+/**
+ * @brief Register event hook on the end of event processing. The hook would be called last.
+ *
+ * The event hook called after he event is processed.
+ * The hook function should have a form `void hook(const struct event_header *eh)`.
+ * The macro makes sure that the hook provided here is called last.
+ * Only one hook can be registered with this macro.
+ *
+ * @param hook_fn Hook function
+ */
+#define EVENT_HOOK_POSTPROCESS_REGISTER_LAST(hook_fn)                                       \
+	const struct {} __event_hook_postprocess_last_sub_redefined = {};                   \
+	_EVENT_HOOK_POSTPROCESS_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_MARKER_LAST_ELEMENT))
+
+
+/** @brief Initialize the Event Manager.
+ *
+ * @retval 0 If the operation was successful. Error values can be added by overwriting
```

Comment:
There is no `event_manager_trace_event_init` function now

Reply:
Ok